File: DSA/Day3 basic math/palindrome.py
n = int(input("Enter a Number: "))

# Step 1: store the original number
original = n

# Step 2: handle negative numbers
if n < 0:
    print("False")   # negative numbers are not palindrome
else:
    reversed_num = 0

    # Step 3: reverse the number using a loop
    while n > 0:
        digit = n % 10            # get last digit
        reversed_num = reversed_num * 10 + digit   # build reversed number
        n = n // 10               # remove last digit

    # Step 4: check if original and reversed are same
    if original == reversed_num:
        print("True")   # palindrome
    else:
        print("False")  # not palindrome

File: JAVA DSA/loops/Palindrome.java
import java.util.Scanner;

class Palindrome {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a Number: ");
        int num = sc.nextInt();
        int reverse =0;
        int original =num;
        for(;num>0;num=num/10){
            int digit=num%10;
            reverse= reverse*10 + digit;
        }
        if(original==reverse){
            System.out.println("Is Palindrome");
        } else{
            System.out.println("Not A Palindrome");
        }
    }
}